grinding_demo_node:
  ros__parameters:
    experiment_time: 300 # min
    pouse_time_list: 
      - 30
      - 60
      - 90
      - 120
      - 150
      - 180
      - 210 
      - 240
      - 270
      - 300
    grinding_ee_link: "pestle_tip"  
    gathering_ee_link: "spatula_tip"

    # ik and motion planner params 
    joint_trajectory_controller_name: "scaled_pos_joint_traj_controller"
    urdf_name: "ur5e" # support "ur3e" and "ur5e"
    ik_solver: "trac_ik" # only "trac_ik" is supported, "ikfast" will be supported in the future
    motion_planner_id: "RRTConnect" # recommended "TRRT"(highly optimal but long planning time) or "RRTConnect" (low optimal but fast planning)
    planning_time: 20 # planning time to timeout in MoveIt motion planning 
    max_attempts: 100 # number of trial for planning by JTC

    # grinding params
    grinding_pos_begining: 
      - -8
      - 0
    grinding_pos_end:
      - -8
      - 0.0001
    grinding_center_pos:
      - 0
      - 0
    grinding_number_of_rotation: 20
    grinding_sec_per_rotation: 0.5 # grinding_number_of_rotation * grinding_sec_per_rotation < 15
    grinding_number_of_waypoints_per_circle: 50  # 50 is enough
    grinding_angle_scale: 0.3 #0 - 0.3
    grinding_rz_begining: 36
    grinding_rz_end: 36
    grinding_vel_scale: 1
    grinding_acc_scale: 1
    grinding_yaw_bias: 3.14159265359 # rad(pi)
    grinding_joint_difference_limit_for_motion_planning: 0.03

    # circular gathering params 
    gathering_pos_begining: 
      - 30
      - 0
    gathering_pos_end:
      - -22
      - 0.0001
    gathering_number_of_rotation: 5
    gathering_sec_per_rotation: 2
    gathering_angle_scale: 0
    gathering_rz_begining: 35
    gathering_rz_end: 35
    gathering_vel_scale: 0.1
    gathering_acc_scale: 0.1
    gathering_number_of_waypoints_per_circle: 100
    gathering_yaw_bias: 3.14159265359 # rad(pi)
    gathering_joint_difference_limit_for_motion_planning: 0.03  